packages/sanity/src/core/studio/screens/RequestAccessScreen.tsx

```diff
@@ -0,0 +1,200 @@
+/* eslint-disable i18next/no-literal-string,@sanity/i18n/no-attribute-string-literals */
+import {Box, Button, Card, Dialog, Flex, Stack, Text, TextInput, useToast} from '@sanity/ui'
+import {useCallback, useEffect, useState} from 'react'
+import {
+  type CurrentUser,
+  getProviderTitle,
+  LoadingBlock,
+  type SanityClient,
+  useActiveWorkspace,
+} from 'sanity'
+
+import {NotAuthenticatedScreen} from './NotAuthenticatedScreen'
+
+interface AccessRequest {
+  id: string
+  status: 'pending' | 'accepted' | 'declined'
+  resourceId: string
+  resourceType: 'project'
+  createdAt: string
+  updatedAt: string
+  updatedByUserId: string
+  requestedByUserId: string
+  note: string
+}
+
+export function RequestAccessScreen() {
+  const [currentUser, setCurrentUser] = useState<CurrentUser | null>(null)
+  const [client, setClient] = useState<SanityClient | undefined>()
+  const [projectId, setProjectId] = useState<string | undefined>()
+  const toast = useToast()
+
+  const [error, setError] = useState<unknown>(null)
+  const [loading, setLoading] = useState(true)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const [hasPendingRequest, setHasPendingRequest] = useState<boolean | undefined>()
+  const [hasTooManyRequests, setHasTooManyRequests] = useState<boolean | undefined>()
+
+  const [note, setNote] = useState<string>('')
+
+  const {activeWorkspace} = useActiveWorkspace()
+
+  const handleLogout = useCallback(() => {
+    activeWorkspace.auth.logout?.()
+  }, [activeWorkspace])
+
+  // Get config info from active workspace
+  useEffect(() => {
+    const subscription = activeWorkspace.auth.state.subscribe({
+      next: ({client: sanityClient, currentUser: user}) => {
+        // Need to get the client, projectId, and user from workspace
+        // because this screen is outside the SourceContext
+        setProjectId(sanityClient.config().projectId)
+        setClient(sanityClient)
+        setCurrentUser(user)
+      },
+      error: setError,
+    })
+
+    return () => {
+      subscription.unsubscribe()
+    }
+  }, [activeWorkspace])
+
+  // Check if user has a pending
+  // access request for this project
+  useEffect(() => {
+    if (!client || !projectId) return
+    client
+      .request<AccessRequest[] | null>({
+        url: `/access/requests/me`,
+      })
+      .then((requests) => {
+        if (requests && requests?.length) {
+          const pendingRequests = requests.filter(
+            (request) =>
+              // Access request is for this project
+              request.resourceId === projectId &&
+              // Access request is still pending
+              request.status === 'pending' &&
+              // Access request is less than 2 weeks old
+              new Date(request.createdAt).getTime() > Date.now() - 2 * 1000 * 60 * 60 * 24 * 7,
+          )
+          if (pendingRequests.length) setHasPendingRequest(true)
+        }
+      })
+      .catch((err) => {
+        console.error(err)
```

Comment:
Not sure if it's standard practice to send silent/handled errors to the console . In `NotAuthenticatedScreen` is seemed like we just threw an error and crashed the studio if we saw one